core/src/main/java/org/infinispan/container/offheap/SegmentedBoundedOffHeapDataContainer.java

```diff
@@ -173,7 +178,7 @@ private void ensureSize() {
             if (map != null) {
                int hashCode = offHeapEntryFactory.getHashCode(firstAddress);
                // This is always non null
-               entryWriteLock = map.getWriteLock(hashCode);
+               entryWriteLock = map.getWriteLock(hashCode).asWriteLock();
```

Comment:
Did you use `asWriteLock()` just to minimize changes, or is there a deeper reason?

Reply:
Minimize changes: to not complicate the code further. It would require retaining another variable `writeStamp` between the various blocks. Did you think we should still change it?

Reply:
Actually implementing it isn't that much more complex. I am updating.

Reply:
Alright, let's go all-in on `StampedLock` :)